system-tests/e2e-suite/src/test/scala/com/hotels/styx/admin/OriginsPathDuplicationOnReloadSpec.scala

```diff
@@ -0,0 +1,90 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin
+
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+import java.nio.file.{Files, Paths}
+
+import com.google.common.io.Files.createTempDir
+import com.hotels.styx.api.HttpRequest.Builder.post
+import com.hotels.styx.api.messages.HttpResponseStatus.INTERNAL_SERVER_ERROR
+import com.hotels.styx.proxy.backends.file.FileBackedBackendServicesRegistry
+import com.hotels.styx.support.ResourcePaths.fixturesHome
+import com.hotels.styx.support.configuration._
+import com.hotels.styx.{StyxClientSupplier, StyxServer, StyxServerSupport}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{BeforeAndAfterAll, FunSpec, ShouldMatchers}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+class OriginsPathDuplicationOnReloadSpec extends FunSpec
+  with StyxServerSupport
+  with StyxClientSupplier
+  with ShouldMatchers
+  with BeforeAndAfterAll
+  with Eventually {
+
+  val tempDir = createTempDir()
+  val originsOk = fixturesHome(classOf[OriginsPathDuplicationOnReloadSpec], "/conf/origins/origins-correct.yml")
+  val originsDuplicatePath = fixturesHome(classOf[OriginsPathDuplicationOnReloadSpec], "/conf/origins/origins-duplicate-path.yml")
+  val styxOriginsFile = Paths.get(tempDir.toString, "origins.yml")
+  var styxServer: StyxServer = _
+
+  it("Responds with INTERNAL_SERVER_ERROR when the backend-services have a duplicate path") {
+    val fileBasedBackendsRegistry = FileBackedBackendServicesRegistry.create(styxOriginsFile.toString)
+    styxServer = StyxConfig().startServer(fileBasedBackendsRegistry)
+
+    styxServer.isRunning should be(true)
+
+    Files.copy(originsDuplicatePath, styxOriginsFile, REPLACE_EXISTING)
+
+    val resp = decodedRequest(post(styxServer.adminURL("/admin/tasks/origins/reload")).build())
+    resp.status() should be(INTERNAL_SERVER_ERROR)
```

Comment:
Why would we respond with INTERNAL SERVER ERROR when this is fundamentally a client error?

Reply:
It's not a client error. The client sent a valid request for the origins to be reloaded. It was the state of the origin config that was invalid.

I presume you made the same observation when you changed the response code in your PR #108 ?

Reply:
Aah right.